### PR TITLE
test(offline): Add test coverage

### PR DESCRIPTION
**Pull request by damassi**
This PR resolves [MOPLAT-548]

### Description

This adds test coverage to the data portion of the app! Looking at the syncManager code, the only bits we need to test are the parsers. Everything else is sufficiently covered by Relay’s auto-generated types and is in effect logic-less — meaning: if ever anything were to change anywhere with our queries, be it name or variables passed or data returned, it would instantly blow up the type-checker in numerous places, including the syncManager. The contract in the component code is sufficient to cover the contract in the offline sync code. With the parsers however we do transform data, and that is subject to breakage.

(Note this only covers the data portion off offline support, not any of the image sync behaviors.) 

cc @artsy/mobile-platform 

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-548]: https://artsyproduct.atlassian.net/browse/MOPLAT-548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/system/sync/syncManager.ts [hunk 1]
```diff
@@ -547,3 +539,8 @@ export const loadRelayDataFromOfflineCache = (
     resetRelayEnvironment(JSON.parse(data!))
   })
 }
+
+export const tests = {
+  syncResults,
+  parsers,
```
Comment (damassi) [inline src/app/system/sync/syncManager.ts:545]: lets keep internal functions private

File: src/app/system/sync/syncManager.ts [hunk 2]
```diff
@@ -547,3 +539,9 @@ export const loadRelayDataFromOfflineCache = (
     resetRelayEnvironment(JSON.parse(data!))
   })
 }
+
+export const tests = {
```
Comment (pvinis) [inline src/app/system/sync/syncManager.ts:543]: there are a few places we do this https://github.com/artsy/eigen/blob/main/src/palette/Theme.tsx#L325-L329. maybe this naming could work here too? obviously non-blocking.

**Review by rajsam003 — APPROVED** [review]

File: src/app/system/sync/syncManager.ts (hunk 2 above)
Comment (damassi) [reply src/app/system/sync/syncManager.ts:543]: oh nice, I like this pattern! Seeing `export const foo` everywhere can often confuse intent but wrapping privates within a tests export makes it nice and clear

Comment (damassi) [reply src/app/system/sync/syncManager.ts:543]: Will update naming since there's already an established pattern 👍

**Pull request merged**